src/timeseriesflattener/spec_processors/temporal.py

```diff
@@ -148,34 +149,118 @@ def _slice_and_aggregate_spec(
 TemporalSpec = Union[PredictorSpec, OutcomeSpec, BooleanOutcomeSpec]
 
 
+def _get_min_max_from_predictiontime_frame(
+    frame: PredictionTimeFrame,
+) -> tuple[dt.datetime, dt.datetime]:
+    if isinstance(frame.df, pl.LazyFrame):
+        df = frame.df.collect()
+    else:
+        df = frame.df
+
+    start_date = df.select(pl.col(frame.timestamp_col_name).min()).item()
+
+    end_date = df.select(pl.col(frame.timestamp_col_name).max()).item()
+
+    return start_date, end_date
+
+
+def _get_longest_lookperiod(lookperiods: list[LookPeriod]) -> dt.timedelta:
+    max_lookperiod = list()
+
+    for lookperiod in lookperiods:
+        max_lookperiod.append((lookperiod.first - lookperiod.last).days)
+
+    max_lookperiod.sort()
+
+    return dt.timedelta(max_lookperiod[0])
+
+
+def _create_date_range(
+    start_date: dt.datetime, end_date: dt.datetime, timedelta: int
+) -> list[dt.datetime]:
+    n = int((end_date - start_date).days / timedelta)
+    return [start_date + dt.timedelta(timedelta * i) for i in range(n + 2)]
```

Comment:
had to add 2 instead of 1 to include the last date, because 365 is not long enough for the average year, so the end was half a day short